ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java

```diff
@@ -1153,13 +1154,31 @@ public List<AmbryReplica> getReplicaIdsByState(AmbryPartition partition, Replica
         RoutingTableSnapshot routingTableSnapshot =
             clusterMapConfig.clusterMapUseAggregatedView ? globalRoutingTableSnapshotRef.get()
                 : dcToRoutingTableSnapshotRef.get(dc).get();
+        Collection<String> resourcesFromRoutingTable = routingTableSnapshot.getResources();
         routingTableSnapshot.getInstancesForResource(resourceName, partition.toPathString(), state.name())
             .stream()
             .map(instanceConfig -> instanceNameToAmbryDataNode.get(instanceConfig.getInstanceName()))
             .filter(dataNode -> dataNode.getDatacenterName().equals(dc))
             .map(dataNode -> ambryDataNodeToAmbryReplicas.get(dataNode).get(partition.toPathString()))
             .filter(Objects::nonNull)
             .forEach(replicas::add);
+        if (replicas.isEmpty()) {
```

Comment:
Nice Idea! One doubt. Since `replicas` is a set of replicas from all dcs, we may not know if replicas in a specific dc returned empty list. Should we have a new set like `currDCReplicas` and check if it is not empty to enter below logic?

Reply:
Comparing with the size of the list now.